packages/web/components/Algolia/rangeSliderWithPanel.js

```diff
@@ -0,0 +1,239 @@
+					left: `${percent}%`,
+					top: '3.2rem',
+					width: 40,
+					height: 25,
+					transform: 'translate(-50%, -100%)',
+					cursor: disabled ? 'not-allowed' : 'grab',
+					zIndex: 1,
+				}}
+				aria-hidden
+				{...getHandleProps(id)}
+			/>
+			<div
+				role="slider"
+				className="slider-handle"
+				aria-valuemin={min}
+				aria-valuemax={max}
+				aria-valuenow={value}
+				style={{
+					left: `${percent}%`,
+					cursor: disabled ? 'not-allowed' : 'grab',
+				}}
+				{...getHandleProps(id)}
+			/>
+		</>
+	);
+}
+
+Handle.propTypes = {
+	domain: PropTypes.arrayOf(PropTypes.number).isRequired,
+	handle: PropTypes.shape({
+		id: PropTypes.string,
+		value: PropTypes.number,
+		percent: PropTypes.number,
+	}).isRequired,
+	disabled: PropTypes.bool,
+	getHandleProps: PropTypes.func.isRequired,
+};
+
+Handle.defaultProps = {
+	disabled: false,
+};
+
+const RangeSliderWithPanel = ({ min, max, refine, currentRefinement, canRefine, header }) => {
+	const [ticksValues, setTicksValues] = useState([currentRefinement.min, currentRefinement.max]);
+
+	useEffect(() => {
+		setTicksValues([currentRefinement.min, currentRefinement.max]);
+	}, [currentRefinement]);
+
+	const onChange = (values) => {
+		refine({ min: values[0] || min, max: values[1] || max });
+	};
+
+	if (!canRefine || ticksValues[0] === undefined || ticksValues[1] === undefined) {
+		return null;
+	}
+
+	const disabledSlider = !canRefine || min === max;
+
+	return (
+		<Panel header={header}>
+			<StyledSlider
+				mode={2}
+				step={1}
+				domain={[min, max]}
+				values={[currentRefinement.min, currentRefinement.max]}
+				disabled={disabledSlider}
+				onChange={onChange}
+				onUpdate={setTicksValues}
+				rootStyle={{
+					position: 'relative',
+					margin: '1.5rem 2.6rem 0',
+					paddingBottom: '4.8rem',
+				}}
+				className="ais-RangeSlider"
+			>
+				<Rail>
+					{({ getRailProps }) => <div className="slider-rail" {...getRailProps()} />}
+				</Rail>
+
+				<Tracks left={false} right={false}>
+					{({ tracks, getTrackProps }) => (
+						<div>
+							{tracks.map(({ id, source, target }) => (
+								<div
+									key={id}
+									className="slider-track"
+									style={{
+										left: `${source.percent}%`,
+										width: `${target.percent - source.percent}%`,
+									}}
+									{...getTrackProps()}
+								/>
+							))}
+						</div>
+					)}
+				</Tracks>
+
+				<Handles>
+					{({ handles, getHandleProps }) => (
+						<div>
+							{handles.map((handle) => (
+								<Handle
+									key={handle.id}
+									handle={handle}
+									domain={[min, max]}
+									getHandleProps={getHandleProps}
+									disabled={disabledSlider}
+								/>
+							))}
+						</div>
+					)}
+				</Handles>
+
+				<Ticks values={ticksValues}>
+					{({ ticks }) => (
+						<div>
+							{ticks.map(({ id, value, percent }, index) => (
+								<div
+									// eslint-disable-next-line react/no-array-index-key
+									key={`${id}_${index}`}
```

Comment:
Why not using just the `id` instead id+ index?

Reply:
Hi @lcnogueira 

Adoidado duplicated key if min and max value is the same.

The id here isn't a trully identifier but just the number